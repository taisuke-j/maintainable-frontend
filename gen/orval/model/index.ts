export * from './patchUserBody';
export * from './postUserBody';
export * from './user';