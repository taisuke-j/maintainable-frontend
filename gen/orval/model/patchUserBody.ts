/**
 * Generated by orval v6.10.3 üç∫
 * Do not edit manually.
 * User API
 * Example API generated by Stoplight
 * OpenAPI spec version: 1.0
 */

export type PatchUserBody = {
  firstName?: string;
  lastName?: string;
  /** If a new email is given, the user's email verified property will be set to false. */
  email?: string;
  dateOfBirth?: string;
};
