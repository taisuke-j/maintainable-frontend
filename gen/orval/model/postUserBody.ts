/**
 * Generated by orval v6.10.3 üç∫
 * Do not edit manually.
 * User API
 * Example API generated by Stoplight
 * OpenAPI spec version: 1.0
 */

export type PostUserBody = {
  firstName: string;
  lastName: string;
  email: string;
  dateOfBirth: string;
};
