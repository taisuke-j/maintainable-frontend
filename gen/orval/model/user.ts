/**
 * Generated by orval v6.10.3 üç∫
 * Do not edit manually.
 * User API
 * Example API generated by Stoplight
 * OpenAPI spec version: 1.0
 */

export interface User {
  /** Unique identifier for the given user. */
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  dateOfBirth?: string;
  /** Set to true if the user's email has been verified. */
  emailVerified: boolean;
  /** The date that the user was created. */
  createDate?: string;
}
