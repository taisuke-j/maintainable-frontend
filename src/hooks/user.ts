import { getGetUserMock } from '../../gen/orval/default/default.msw';

// Add additional logics here if needed
// TODO: Currently just using the mock
export const useGetUser = (_userId: number) => getGetUserMock();