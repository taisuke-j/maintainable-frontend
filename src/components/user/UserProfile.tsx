import { User } from "../../../gen/orval/model/user";

export const UserProfile = (props: User) => {
  return (
    <>
      <h1>{props.firstName}</h1>
      <p>Email: {props.email}</p>
    </>
  );
}