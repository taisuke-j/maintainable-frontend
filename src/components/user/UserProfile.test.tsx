import { render } from "@testing-library/react";

import { UserProfile } from "./UserProfile";
import { User } from "../../../gen/orval/model/user";

describe("Simple working test", () => {
  test("should render correctly", () => {
    const user: User = {
      id: 1,
      firstName: "Tom",
      lastName: "Smith",
      email: "tom@example.com",
      emailVerified: true,
    };

    const { container } = render(<UserProfile {...user} />);
    expect(container).toMatchSnapshot();
  });
});
