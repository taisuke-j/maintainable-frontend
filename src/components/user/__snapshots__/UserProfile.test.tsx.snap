// Vitest Snapshot v1

exports[`Simple working test > should render correctly 1`] = `
<div>
  <h1>
    Tom
  </h1>
  <p>
    Email: 
    tom@example.com
  </p>
</div>
`;
